### PR TITLE
improve: various add-ons: SAST (sonar) Anonymous inner class lambda

### PR DESCRIPTION
**Pull request by kingthorin**
- Replace anonymous inner classes (runnables, listeners, comparators, equalizers, etc) with lambdas.
- CHANGELOGs updated with maintenance note where warranted.

Signed-off-by: kingthorin <kingthorin@users.noreply.github.com>

### CODE REVIEW

File: addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java [hunk 1]
```diff
@@ -458,12 +446,8 @@ protected JCheckBox getEnabledCheckBox() {
...
             enabledCheckBox.addActionListener(
-                    new ActionListener() {
-
-                        @Override
-                        public void actionPerformed(ActionEvent e) {
-                            setButtonStates();
-                        }
+                    e -> {
+                        setButtonStates();
                     });
```
Comment (thc202) [inline addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java:451]: `enabledCheckBox.addActionListener(e -> setButtonStates());`

Comment (thc202) [reply addOns/alertFilters/src/main/java/org/zaproxy/zap/extension/alertFilters/DialogAddAlertFilter.java]: This applies to all cases where there is only a single statement.

**Comment by kingthorin** [comment]
Tweaked

File: addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/WebSocketProxy.java [hunk 2]
```diff
@@ -891,11 +887,8 @@ public void removeSenderListener(WebSocketSenderListener senderListener) {
     private static synchronized Comparator<WebSocketSenderListener> getSenderListenersComparator() {
         if (null == senderListenersComparator) {
             senderListenersComparator =
-                    new Comparator<WebSocketSenderListener>() {
-                        @Override
-                        public int compare(WebSocketSenderListener o1, WebSocketSenderListener o2) {
-                            return Integer.compare(o1.getListenerOrder(), o2.getListenerOrder());
-                        }
+                    (WebSocketSenderListener o1, WebSocketSenderListener o2) -> {
```
Comment (thc202) [inline addOns/websocket/src/main/java/org/zaproxy/zap/extension/websocket/WebSocketProxy.java:890]: This one was not tweaked (the types can be removed as well, same in other places).

File: addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/ZestResultsTableModel.java [hunk 3]
```diff
@@ -71,12 +71,8 @@ public ZestResultsTableModel() {
     public void clear() {
         if (View.isInitialised() && !EventQueue.isDispatchThread()) {
             EventQueue.invokeLater(
-                    new Runnable() {
-
-                        @Override
-                        public void run() {
-                            clear();
-                        }
+                    () -> {
```
Comment (thc202) [inline addOns/zest/src/main/java/org/zaproxy/zap/extension/zest/ZestResultsTableModel.java:74]: Same here (and following).

**Comment by thc202** [comment]
There are still more cases though.

**Comment by thc202** [comment]
Re the method references, they could be replaced at the same time (there are not that many, ~20).

**Comment by thc202** [comment]
Re `ExtensionWappalyzer` comment, I'd suggest extract a method and keep the lambda lean.

**Comment by kingthorin** [comment]
> 
> 
> Re `ExtensionWappalyzer` comment, I'd suggest extract a method and keep the lambda lean.

Addressed in db5d54c.

**Comment by thc202** [comment]
I was suggesting moving everything and use method reference.

**Comment by kingthorin** [comment]
Could do it like this:
```java
    @Override
    public void postInstall() {
        super.postInstall();
        if (getView() != null) {
            // Set true to test icon rendering
            EventQueue.invokeLater(() -> doPostInstall(false));
        }
    }

    private void doPostInstall(boolean renderIcons) {
        getTechPanel().setTabFocus();
        if (renderIcons) {
            getApplications().forEach(app -> addApplicationsToSite("http://localhost",
                    new ApplicationMatch(app)));
        }
    }
```

But I don't see how it can be a method reference.

**Comment by thc202** [comment]
I was being literal when saying to move everything, e.g.:
```Java
private void focusTab() {
    getTechPanel().setTabFocus();
    // Un-comment to test icon rendering
    /*
     * getApplications() .forEach( app -> addApplicationsToSite( "http://localhost",
     * new ApplicationMatch(app)));
     */
}
``

**Comment by thc202** [comment]
Regarding the latest changes, it seems the `ExtensionPayloader` classes are still using braces.

**Review by thc202 — APPROVED** [review]

**Comment by thc202** [comment]
Thank you!

**Review by psiinon — APPROVED** [review]

**Pull request merged**